src/main/java/com/microsoft/sqlserver/jdbc/ConfigurableRetryLogic.java

```diff
@@ -0,0 +1,206 @@
+                String[] arr = rule.getError().split(COMMA);
+
+                for (String retryError : arr) {
+                    ConfigurableRetryRule splitRule = new ConfigurableRetryRule(retryError, rule);
+                    stmtRules.put(Integer.parseInt(splitRule.getError()), splitRule);
+                }
+            } else {
+                stmtRules.put(Integer.parseInt(rule.getError()), rule);
+            }
+        }
+    }
+
+    private static String getCurrentClassPath() throws SQLServerException {
+        String location = "";
+        String className = "";
+
+        try {
+            className = new Object() {}.getClass().getEnclosingClass().getName();
+            location = Class.forName(className).getProtectionDomain().getCodeSource().getLocation().getPath();
+            location = location.substring(0, location.length() - 16);
+            URI uri = new URI(location + FORWARD_SLASH);
+            return uri.getPath() + DEFAULT_PROPS_FILE; // For now, we only allow "mssql-jdbc.properties" as file name.
+        } catch (URISyntaxException e) {
+            MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_AKVURLInvalid"));
+            Object[] msgArgs = {location + FORWARD_SLASH};
+            throw new SQLServerException(form.format(msgArgs), null, 0, e);
+        } catch (ClassNotFoundException e) {
+            MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_UnableToFindClass"));
+            Object[] msgArgs = {className};
+            throw new SQLServerException(form.format(msgArgs), null, 0, e);
+        }
+    }
+
+    private static LinkedList<String> readFromFile() throws SQLServerException {
+        String filePath = getCurrentClassPath();
+        LinkedList<String> list = new LinkedList<>();
+
+        try {
+            File f = new File(filePath);
+            try (BufferedReader buffer = new BufferedReader(new FileReader(f))) {
+                String readLine;
+                while ((readLine = buffer.readLine()) != null) {
+                    if (readLine.startsWith(RETRY_EXEC)) {
+                        String value = readLine.split(EQUALS_SIGN)[1];
+                        Collections.addAll(list, value.split(SEMI_COLON));
+                    }
+                }
+            }
+            timeLastModified = f.lastModified();
+        } catch (FileNotFoundException e) {
+            // If the file is not found either A) We're not using CRL OR B) the path is wrong. Do not error out, instead
+            // log a message.
+            if (CONFIGURABLE_RETRY_LOGGER.isLoggable(java.util.logging.Level.FINER)) {
+                CONFIGURABLE_RETRY_LOGGER.finest("No properties file exists or the file path is incorrect.");
+            }
+        } catch (IOException e) {
+            MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_errorReadingStream"));
+            Object[] msgArgs = {e.toString()};

```

Comment:
this seems to be the wrong error msg used, also would be good to add the file path to the error msg

Reply:
An IO Exception is thrown if the above BufferedReader was interrupted during read. This appears to be the correct error for this case, and is used in our driver in other cases where we have an potential interrupted reader. What should be used instead?

Reply:
My mistake, fixed.